src/Core/Config/ConfigCache.php

```diff
@@ -0,0 +1,183 @@
+<?php
+
+namespace Friendica\Core\Config;
+
+/**
+ * The Friendica config cache for the application
+ * Initial, all *.config.php files are loaded into this cache with the
+ * ConfigCacheLoader ( @see ConfigCacheLoader )
+ *
+ * Is used for further caching operations too (depending on the ConfigAdapter )
+ */
+class ConfigCache implements IConfigCache, IPConfigCache
+{
+	/**
+	 * NEVER, EVER use this public config array outside of the class
+	 * It is only public due to backward compatibility to .htconfig.php
+	 *
+	 * @var array The cached config array
+	 */
+	public $config;
+
+	/**
+	 * @param array $config    A initial config array
+	 */
+	public function __construct($config = [])
+	{
+		$this->config = [];
+
+		if (isset($config)) {
+			$this->loadConfigArray($config, true);
```

Comment:
This line equals `$this->config = $config`, which is what this constructor should boil down to.